### DISCUSSION TITLE
Navigation on close does not incude any result anymore.

### DISCUSSION THREAD

**Discussion by scchauhan** [MvvmCross/MvvmCross#4843] [General]
Hi,
I am using .Net Maui (Upgraded) ios code and its using MVVMCross(latest) but Navigation has been change in latest dll, I need to return the ViewModel on close to previous page. 
Can some one help me in that? along with that I am concern about bind tap, bindvisble properties as well.
Appreciate your help.

**Comment by Cheesebaron** [comment]
Are you referring to `TResult` when closing a ViewModel? This was removed 2 years ago, see #4312. This functionality is very broken, and I opted to remove it.

You are not the first person asking about this, there are various ways to return similar functionality. 

- Some people use `IMvxMessenger` to publish an event when navigating backwards
- Some are using callbacks in the parameter you pass to the ViewModel when navigating
- Some are using scope objects and persisting that between ViewModels somehow

It is essentially up to you how you want to approach it. You can also fairly easily restore the functionality by adding back the methods for this in your own NavigationService implementation and extend ViewModel implementation.
